.circleci/config/commands/go_test.yml

```diff
@@ -116,8 +117,14 @@ steps:
           docker cp . testcontainer:/go/src/github.com/hashicorp/vault/
           docker cp $DOCKER_CERT_PATH/ testcontainer:$DOCKER_CERT_PATH
 
+          # Copy the downloaded modules inside the container.
+          docker exec testcontainer sh -c 'mkdir -p /go/pkg'
+          docker cp "$(go env GOPATH)/pkg/mod" testcontainer:/go/pkg/mod
+
           docker exec -w /go/src/github.com/hashicorp/vault/ \
-            -e CIRCLECI -e GOCACHE=/tmp/gocache -e VAULT_CI_GO_TEST_RACE \
+            -e CIRCLECI -e VAULT_CI_GO_TEST_RACE \
+            -e GOCACHE=/tmp/gocache \
+            -e GOPROXY="off" \
```

Comment:
Why no GOPRIVATE?

Reply:
Good point, I intended to replace this file with the ENT one. I'll open a PR to that effect now @ncabatoff

Reply:
Please see https://github.com/hashicorp/vault/pull/11951 and its sister ent PR